packages/wdio-browserstack-service/src/service.js

```diff
@@ -57,32 +61,50 @@ export default class BrowserstackService {
             `${test.parent} - ${test.title}`
         )
 
-        if (!test.passed) {
-            this.failures++
-            this.failReason = (test.error && test.error.message ? test.error.message : 'Unknown Error')
+        if (!passed) {
+            this.failReasons.push(error && error.message || 'Unknown Error')
         }
     }
 
-    after() {
-        return this._update(this.sessionId, this._getBody())
+    after(result) {
+        if(this.preferScenarioName && this.scenariosThatRan.length === 1){
```

Comment:
can we add a comment block documenting what this is doing?

Reply:
How's this?

Reply:
👍